src/Features/Core/Portable/Completion/CompletionItemRules.cs

```diff
@@ -69,25 +122,30 @@ public sealed class CompletionItemRules
         /// <param name="enterKeyRule">Rule about whether the enter key is passed through to the editor after the selected item has been committed.</param>
         /// <param name="formatOnCommit">True if the modified text should be formatted automatically.</param>
         /// <param name="matchPriority">True if the related completion item should be initially selected.</param>
+        /// <param name="selectionBehavior">How this item should be selected if no text has been typed after the completion list is brought up.</param>
         /// <returns></returns>
         public static CompletionItemRules Create(
             ImmutableArray<CharacterSetModificationRule> filterCharacterRules = default(ImmutableArray<CharacterSetModificationRule>),
             ImmutableArray<CharacterSetModificationRule> commitCharacterRules = default(ImmutableArray<CharacterSetModificationRule>),
             EnterKeyRule enterKeyRule = EnterKeyRule.Default,
             bool formatOnCommit = false,
-            int? matchPriority = null)
+            int? matchPriority = null,
+            CompletionItemSelectionBehavior selectionBehavior = CompletionItemSelectionBehavior.Default)
```

Comment:
This is the new factory constructor with all optional values.  The old factory method is there for binary compat.  But could that is recompiled will call into this constructor instead.